app/models/agents/webhook_agent.rb

```diff
@@ -53,7 +53,16 @@ def default_options
     end
 
     def receive_web_request(request)
-      params = request.params.except(:action, :controller, :agent_id, :user_id, :format)
+      # check the secret
+      secret = request.path_parameters[:secret]
+      return ["Not Authorized", 401] unless secret == interpolated['secret']
+
+      params = request.query_parameters.dup
+      begin
+        params.update(request.request_parameters)
```

Comment:
Just double checking, is that the same order Rails does the merge of the two parameter types?

Reply:
Yes, POST parameters take precedence over GET parameters like any other framework since the CGI era.